samples/cpp/digits.cpp

```diff
@@ -1,375 +1,187 @@
+    "{ height      | 480 | Set the height of the camera }"
+    "{ thr         | 0.8 | Confidence threshold. }";
 
-static void help(char** argv)
-{
-    cout <<
-    "\n"
-    "SVM and KNearest digit recognition.\n"
-    "\n"
-    "Sample loads a dataset of handwritten digits from 'digits.png'.\n"
-    "Then it trains a SVM and KNearest classifiers on it and evaluates\n"
-    "their accuracy.\n"
-    "\n"
-    "Following preprocessing is applied to the dataset:\n"
-    " - Moment-based image deskew (see deskew())\n"
-    " - Digit images are split into 4 10x10 cells and 16-bin\n"
-    "   histogram of oriented gradients is computed for each\n"
-    "   cell\n"
-    " - Transform histograms to space with Hellinger metric (see [1] (RootSIFT))\n"
-    "\n"
-    "\n"
-    "[1] R. Arandjelovic, A. Zisserman\n"
-    "    \"Three things everyone should know to improve object retrieval\"\n"
-    "    http://www.robots.ox.ac.uk/~vgg/publications/2012/Arandjelovic12/arandjelovic12.pdf\n"
-    "\n"
-    "Usage:\n"
-    << argv[0] << endl;
-}
-
-static void split2d(const Mat& image, const Size cell_size, vector<Mat>& cells)
-{
-    int height = image.rows;
-    int width = image.cols;
+// Find best class for the blob (i.e. class with maximal probability)
+static void getMaxClass(const Mat &probBlob, int *classId, double *classProb);
 
-    int sx = cell_size.width;
-    int sy = cell_size.height;
-
-    cells.clear();
-
-    for (int i = 0; i < height; i += sy)
-    {
-        for (int j = 0; j < width; j += sx)
-        {
-            cells.push_back(image(Rect(j, i, sx, sy)));
-        }
-    }
-}
+void predictor(dnn::Net net, Mat &roi, int &class_id, double &probability);
 
-static void load_digits(const char* fn, vector<Mat>& digits, vector<int>& labels)
+int main(int argc, char **argv)
 {
-    digits.clear();
-    labels.clear();
-
-    String filename = samples::findFile(fn);
-
-    cout << "Loading " << filename << " ..." << endl;
-
-    Mat digits_img = imread(filename, IMREAD_GRAYSCALE);
-    split2d(digits_img, Size(SZ, SZ), digits);
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
 
-    for (int i = 0; i < CLASS_N; i++)
+    if (argc == 1 || parser.has("help"))
     {
-        for (size_t j = 0; j < digits.size() / CLASS_N; j++)
-        {
-            labels.push_back(i);
-        }
+        parser.printMessage();
+        return 0;
     }
-}
 
-static void deskew(const Mat& img, Mat& deskewed_img)
-{
-    Moments m = moments(img);
+    int vWidth = parser.get<int>("width");
+    int vHeight = parser.get<int>("height");
+    float confThreshold = parser.get<float>("thr");
+    std::string modelTxt = parser.get<String>("modelTxt");
+    std::string modelBin = parser.get<String>("modelBin");
 
-    if (abs(m.mu02) < 0.01)
+    dnn::Net net;
+    try
     {
-        deskewed_img = img.clone();
-        return;
+        net = dnn::readNetFromCaffe(modelTxt, modelBin);
```

Comment:
Better to use `readNet` just in case of one wanted to run this sample with non-Caffe model